### ISSUE TITLE
CommonsChunkPlugin does not move shared class

### ISSUE THREAD

**Issue by rasentry** [webpack/webpack#3981]
**Do you want to request a *feature* or report a *bug*?**
Feature

**What is the current behavior?**
### Class hierarchy
```
            Main (entry, common chunk)
           /    \
     ClassA      ClassB
       \          \
        \          ClassC
         \         /
       ReusableComponent
```
Main require ClassA and ClassB using require.ensure().
ClassB require ClassC using require.ensure().
ClassA and ClassC require ReusableComponent using simple require().
<p>
Finaly webpack emits following chunks:
<ul>
<li>0.js - contains ClassA, <b>ReusableComponent</b></li>
<li>1.js - contains ClassB</li>
<li>2.js - contains ClassC, <b>ReusableComponent</b></li>
<li>main.js - contains Main class</b></li>
</ul>
</p>

### Problem
I try to configure CommonsChunkPlugin to move ReusableClass to main chunk in order to remove code duplication.
```javascript
new webpack.optimize.CommonsChunkPlugin({
            name: 'main',
            minChunks: 2,
            children: true
        })
```
And webpack does not do this.

**If the current behavior is a bug, please provide the steps to reproduce.**
repo: https://github.com/rasentry/webpack-chunks

**What is the expected behavior?**
ReusableCompoent class has to move to chunk main.

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
Node.js 7.2.1
Windows 10

**Comment by sokra** [comment]
It's expected this way, as `children: true` only affects direct children of the chunk and the chunk with ClassC is not a direct child of main.

Maybe it makes sense to add a recursive mode to grab also modules from indirect children.

**Comment by rasentry** [comment]
@sokra thanks for you answer.
If I need to create new feature request or maybe possible to convert this issue to feature request?

**Comment by EvNaverniouk** [comment]
I'm having the same issue. I've created a detailed explanation here: http://stackoverflow.com/questions/41625147/how-to-control-chunking-of-shared-code-using-import-chunks-webpack-2-2

And also a tiny repo to reproduce this issue here: https://github.com/EvNaverniouk/webpack-code-splitting

**Comment by Tom910** [comment]
The same problem. Is it worth waiting for a fix?

**Comment by rasentry** [comment]
@Tom910 waiting for PR #4416 and #4255

**Comment by iNikNik** [comment]
Workaround: https://github.com/webpack/webpack/issues/5109#issuecomment-311154920

**Comment by bebraw** [comment]
Looks like this was resolved in #5764.

**Comment by mistrykaran91** [comment]
Can I exclude any specific user created chunk from ng build --prod, currently I'm having user & dashboard module, I want to exclude "user" module in ng buid --prod ? How can I achieve that?
